contracts/src/systems/actions.cairo

```diff
@@ -100,21 +100,45 @@ mod actions {
             let player: ContractAddress = get_caller_address();
 
             mancala_game.validate_move(player, selected_pit);
-            let (mut current_player, mut opponent) = mancala_game.get_players(world);
-            // Get seeds from the selected pit and validate it's not empty
-            let mut seeds = mancala_game.get_seeds(current_player, selected_pit);
-            if seeds == 0 {
-                panic!("Selected pit is empty. Choose another pit.");
-            }
-            mancala_game.clear_pit(ref current_player, selected_pit);
-            mancala_game
-                .distribute_seeds(ref current_player, ref opponent, ref seeds, selected_pit);
-            if mancala_game.is_game_finished(current_player, opponent) {
-                mancala_game.set_winner(current_player, opponent);
+
+            if mancala_game.status == GameStatus::InProgress {
```

Comment:
This if statement is not needed and should be removed. We are asserting the value of InProgress on line 94

Reply:
mmh i think we need it : 

```
mancala_game.validate_move(player, selected_pit);
```

this line calls `validate_move` and has the potential to modify `status` for `TimeOut` : 

```
    fn validate_move(ref self: MancalaGame, player: ContractAddress, selected_pit: u8) {
        if player != self.current_player {
            panic!("You are not the current player");
        }
        if selected_pit < 1 || selected_pit > 6 {
            panic!("Invalid pit, choose between 1 and 6");
        }

        let execution_info = get_execution_info_syscall().unwrap_syscall().unbox();
        let block_info = execution_info.block_info.unbox();

        if player == self.player_one && self.last_move > block_info.block_number
            + self.time_between_move {
            self.winner = self.player_two;
            self.status = GameStatus::TimeOut;
        }
        if player == self.player_two && self.last_move > block_info.block_number
            + self.time_between_move {
            self.winner = self.player_one;
            self.status = GameStatus::TimeOut;
        }

        self.last_move = block_info.block_number;
    }
```

In this case, we dont want the tx to revert and we just do : 
```
set!(world, (mancala_game));
(mancala_game.current_player, mancala_game.status)
```

Reply:
Yes but we assert this at line 94: assert!(mancala_game.status == GameStatus::InProgress, "Game is not in progress");

In this case if the game is not in progress the transaction will revert.

Reply:
And I think we should add this to the validate_move fn for code readability

`
assert!(mancala_game.status == GameStatus::InProgress, "Game is not in progress");
            assert!(
                mancala_game.player_two != ContractAddressZeroable::zero(),
                "Player two not yet set."
            );
`

Reply:
No i mean : 
- we assert at line 94: assert!(mancala_game.status == GameStatus::InProgress, "Game is not in progress");
- then we call validate_move , which might modify mancal_game statuts to TimeOut
- then, we execute the content of the move only if the status is still InProgress, not Timeout 

does that make sense?

Reply:
this allows to have 2 ways to trigger a timeout: 
- the player whos turn in the next, but trying to move too late
- any player calling `time_out`  function without the intention to make a move

Reply:
Ok yes I see and did not notice your code update within the validate_move. I will approve this pr however I would also like to note that for readability it might be better to abstract the timeout logic into its own function called validate_timeout.

Reply:
done!